lib/ffmprb/file.rb

```diff
@@ -0,0 +1,184 @@
+require 'json'
+require 'mkfifo'
+require 'tempfile'
+
+module Ffmprb
+
+  class File
+
+    class << self
+
+      def buffered_fifo(extname='.tmp')
+        input_fifo_file = temp_fifo(extname)
+        output_fifo_file = temp_fifo(extname)
+
+        Ffmprb.logger.debug "Opening #{input_fifo_file.path}>#{output_fifo_file.path} for buffering"
+        buff = Util::IoBuffer.new(
+          ->{
+            Ffmprb.logger.debug "Trying to open #{input_fifo_file.path} for reading+buffering"
+            ::File.open(input_fifo_file.path, 'r')
+          },
+          ->{
+            Ffmprb.logger.debug "Trying to open #{output_fifo_file.path} for buffering+writing"
+            ::File.open(output_fifo_file.path, 'w')
+            }
+          )
+        thr = Util::Thread.new do
+          buff.flush!
+          Ffmprb.logger.debug "IoBuffering from #{input_fifo_file.path} to #{output_fifo_file.path} ended"
+          input_fifo_file.remove
+          output_fifo_file.remove
+        end
+        Ffmprb.logger.debug "IoBuffering from #{input_fifo_file.path} to #{output_fifo_file.path} started"
+
+        yield buff  if block_given?  # XXX a hidden option
+
+        OpenStruct.new in: input_fifo_file, out: output_fifo_file, thr: thr
+      end
+
+      def create(path)
+        new(path: path, mode: :write).tap do |file|
+          Ffmprb.logger.debug "Created file with path: #{file.path}"
+        end
+      end
+
+      def open(path)
+        new(path: (path.respond_to?(:path)? path.path : path), mode: :read).tap do |file|
```

Comment:
This is `Ffmprb::File`, we set the rules — for convenience: here we can create a `Ffmprb::File` off a path string, or a ruby `File`, or even another `Ffmprb::File`. It's convenient. If usecases aren't clear, I'll elaborate.